src/main/java/de/retest/surili/commons/core/TestSuite.java

```diff
@@ -18,6 +21,16 @@
 	 */
 	Set<TestCase> getTestCases();
 
+	/**
+	 * @implNote Sorted alphabetically according to {@link #getName()}.
+	 * @return {@link #getTestCases()} as a list.
+	 */
+	default List<TestCase> getTestCasesAsList() {
```

Comment:
Do you need it sorted? Otherwise you can depend on the order of the `Set` and use a `TreeSet`.

Reply:
I need a guaranteed order, e.g., for tests. But I cannot guarantee what kind of `Set` is going to be used. Technically, we could try to enforce `SortedSet`, but this approach doesn't feel right to me. A test suite shouldn't have a fixed order, only on demand.

What's your say on this?

Reply:
> A test suite with a set (i.e. unordered) of TestCases.

I mean, you know the code and its usage better than me. I was just wondering, if the overhead to sort the set is worth it. If yes, go ahead (as `Lists`&mdash;per definition&mdash;assume a general ordering and you always need to sort via `#getName()`).

I typically find myself converting a `Set` to a `List` because I need the access to the index while I generally do not care about the exact content (i.e. the order). In this case, I would suggest to keep this implementation/api as simple as possible and sort on demand and in place.

In tests you can easily provide a `SortedSet` if needed.

Reply:
Unfortunately, I cannot provide a `SortedSet` in integration tests as I'm not in control over the test suite sets. I think for now, this is the approach I need.